kagiso_smart_404/utils.py

```diff
@@ -1,17 +1,37 @@
+import re
+
 from wagtail.wagtailcore.models import Page
 
 
+def determine_if_slug_matches_one_page_exactly(slug, root_page):
+    # Regex Voodoo right here:
+    # input =/my/url/, slug=url
+
+    result = re.match('/.*/(.*)/', slug)
+    if result is None:
+        # input =/url/, slug=url
+        # For urls passed in with out path depth
+        result = re.match('/(.*)/', slug)
+
+    cleaned_slug = result.group(1)
+    try:
+        return Page.objects.descendant_of(root_page).get(
+            slug=cleaned_slug)
```

Comment:
as by definition if the slug is misspelled you cant do a look up by slug, hence the use of trigrams